addons/mass_mailing/models/mailing.py

```diff
@@ -1335,6 +1342,52 @@ def _image_to_url(b64image: bytes):
                 node.attrib['src'] = _image_to_url(image)
                 modified = True
 
+    def _crop_vml_images(self, body_html):
+        """
+        Find VML v:image elements, crop their source images, make an attachement
+        out of them and replace their source with an url to the attachement.
+        """
+
+        def _image_to_url(b64image: bytes):
+            """Store an image in an attachement and returns an url"""
+            attachment = self.env['ir.attachment'].create({
+                'datas': b64image,
+                'name': "cropped_image_mailing_{}".format(self.id),
+                'type': 'binary',})
+
+            attachment.generate_access_token()
+
+            return '/web/image/%s?access_token=%s' % (
+                attachment.id, attachment.access_token)
+
+        root = lxml.html.fromstring(body_html)
+        modified = False
+        for comment in root.xpath('//comment()'):
+            is_mso = mso_re.match(comment.text)
+            if is_mso:
```

Comment:
My python is a little bit rusty but isn't this equivalent to the following ?
```py
for comment in [c for c in root.xpath('//comment()') if mso_re.match(c.text)]:
```

Not necessarily trying to golf per se, but the construction below already has quite a lot of depth so I figured we might gain a few here.

Reply:
Done.